diff --git a/subsys/debug/thread_info.c b/subsys/debug/thread_info.c
index ca9218d6a67..08074ed7748 100644
--- a/subsys/debug/thread_info.c
+++ b/subsys/debug/thread_info.c
@@ -92,6 +92,10 @@ const size_t _kernel_thread_info_offsets[] = {
 #elif defined(CONFIG_ARCH_POSIX)
 	[THREAD_INFO_OFFSET_T_STACK_PTR] = offsetof(struct k_thread,
 						callee_saved.thread_status),
+#elif defined(CONFIG_WASM)
+	/* The C shadow stack's pointer, saved at every switch. */
+	[THREAD_INFO_OFFSET_T_STACK_PTR] = offsetof(struct k_thread,
+						callee_saved.sp),
 #elif defined(CONFIG_XTENSA)
 /* Xtensa does not store stack pointers inside thread objects.
  * The registers are saved in thread stack where there is
